pkg/cmd/ssh/ssh_patch_options.go

```diff
@@ -0,0 +1,343 @@
+	if err != nil {
+		return nil, err
+	}
+
+	bastion, err := gardenClient.GetBastion(ctx, shoot.Namespace, o.BastionName)
+	if err != nil {
+		return nil, err
+	}
+
+	if bastion.ObjectMeta.UID == "" {
+		return nil, fmt.Errorf("Bastion '%s' in namespace '%s' not found", o.BastionName, shoot.Namespace)
+	}
+
+	return bastion, nil
+}
+
+func getAuthInfo(ctx context.Context, manager target.Manager) (*api.AuthInfo, error) {
+	var authInfo *api.AuthInfo
+
+	sshTarget, _ := manager.CurrentTarget()
+	gardenTarget := target.NewTarget(sshTarget.GardenName(), "", "", "")
+
+	if clientConfig, err := manager.ClientConfig(ctx, gardenTarget); err == nil {
+		if rawConfig, err := clientConfig.RawConfig(); err == nil {
+			context, ok := rawConfig.Contexts[rawConfig.CurrentContext]
+			if !ok {
+				return nil, fmt.Errorf("Could not determine current authInfo. Failed to read current context from kubeconfig")
+			}
+
+			authInfo, ok = rawConfig.AuthInfos[context.AuthInfo]
+			if !ok {
+				return nil, fmt.Errorf("Could not determine current authInfo. Failed to read auth info/user from kubeconfig")
+			}
+		} else {
+			return nil, fmt.Errorf("Could not retrieve raw config: %v", err)
+		}
+	} else {
+		return nil, fmt.Errorf("Could not retrieve client config for target garden %s: %v", sshTarget.GardenName(), err)
+	}
+
+	return authInfo, nil
+}
+
+func getBastionsOfUser(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) ([]*operationsv1alpha1.Bastion, error) {
+	bastionOfUser := []*operationsv1alpha1.Bastion{}
+
+	currentUser, err := getCurrentUser(ctx, gardenClient, authInfo)
+	if err != nil {
+		return nil, err
+	} else if currentUser == "" {
+		return bastionOfUser, nil
+	}
+
+	list, err := gardenClient.ListBastions(ctx)
+	if err != nil || len(list.Items) == 0 {
+		return nil, err
+	}
+
+	for i := range list.Items {
+		bastion := list.Items[i]
+		if createdBy, ok := bastion.Annotations["gardener.cloud/created-by"]; ok {
+			if createdBy == currentUser {
+				bastionOfUser = append(bastionOfUser, &bastion)
+			}
+		}
+	}
+
+	return bastionOfUser, nil
+}
+
+func getBastionNamesOfUser(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) ([]string, error) {
+	bastions, err := getBastionsOfUser(ctx, gardenClient, authInfo)
+	if err != nil {
+		return []string{}, err
+	}
+
+	names := make([]string, len(bastions))
+	for i, bastion := range bastions {
+		names[i] = bastion.Name
+	}
+
+	return names, nil
+}
+
+func patchBastion(ctx context.Context, o *SSHPatchOptions, gardenClient gardenClient.Client, sshTarget target.Target) error {
+	bastion, err := getBastion(ctx, o, gardenClient, sshTarget)
+	if err != nil {
+		return err
+	}
+
+	oldBastion := bastion.DeepCopy()
+
+	hasChanges, _, err := patchBastionResource(o, bastion)
+
+	if err != nil {
+		return err
+	} else if !hasChanges {
+		return nil
+	}
+
+	return gardenClient.PatchBastion(ctx, bastion, oldBastion)
+}
```

Comment:
in addition to my other [comment](https://github.com/gardener/gardenctl-v2/pull/180/files#r1015745823), I would inline most of the extracted functions, as it adds imo unneeded complexity
I have also proposed to change the function name to `patchBastionIngressPolicies`
```go
func patchBastionIngressPolicies(ctx context.Context, o *SSHPatchOptions, gardenClient gardenClient.Client, sshTarget target.Target) error {
	bastion, err := getBastion(ctx, o, gardenClient, sshTarget) // TODO bastion should be taken from the options struct
	if err != nil {
		return err
	}

	oldBastion := bastion.DeepCopy()

	var policies []operationsv1alpha1.BastionIngressPolicy

	for _, cidr := range o.CIDRs {
		if *bastion.Spec.ProviderType == "gcp" {
			ip, _, err := net.ParseCIDR(cidr)
			if err != nil {
				return err // this should never happen, as it is already checked within the Validate function
			}

			if ip.To4() == nil {
				if !o.AutoDetected {
					return fmt.Errorf("GCP only supports IPv4: %s", cidr)
				}

				fmt.Fprintf(o.IOStreams.Out, "GCP only supports IPv4, skipped CIDR: %s\n", cidr)

				continue // skip
			}
		}

		policies = append(policies, operationsv1alpha1.BastionIngressPolicy{
			IPBlock: networkingv1.IPBlock{
				CIDR: cidr,
			},
		})
	}

	if len(policies) == 0 {
		return errors.New("there are no ingress policies to be applied")
	}

	bastion.Spec.Ingress = policies

	return gardenClient.PatchBastion(ctx, bastion, oldBastion)
}
```